Gems/AWSCore/Code/Tools/ResourceMappingTool/utils/json_utils.py

```diff
@@ -119,6 +130,24 @@ def convert_json_dict_to_resources(json_dict: Dict[str, any]) -> List[ResourceMa
     return resources
 
 
+def load_resource_mapping_json_schema(schema_path: str) -> None:
```

Comment:
To me, we have a schema, we should not write our validator as its brittle / reinvents the wheel.

But I understand, we don't have jsonschema available to us. Did we look at cost to bring this in?

Reply:
FYI: there's a concern that even if we brought it in, we'd be stuck using an [older `json-schema` version](https://json-schema.org/obsolete-implementations#validators) on the C++ side (4) than what [`jsonschema` supports](https://json-schema.org/implementations.html#validator-python) (listed as 6, which from the docs sounds like it represents a big gap). 

But curious to know if any testing was done re: the effects of using it anyway & what it would take w/r/t dependencies.

Reply:
I think we have the discussion in the slack already, but I can have a summary here:
1. c++ is using rapidjson which supports draft-4 version only (which is the top priority we should satisfy)
2. @allisaurus has pointed out python jsonschema supports draft-6 (see difference here https://json-schema.org/draft-06/json-schema-release-notes.html), if it is true, then we can do use it, as there is backward incompatible change we are currently using. But independently, we need to verify whether jsonschema can work with draft-4
3. if we can verify number 2, then we can work with core team and discuss whether we can add jsonschema as part of o3de python env
[UX behavior] 4. resource mapping tool is not simply a thing using lib schema validation, as it is an editor, it should indicate the errors, not just fail file validation against schema.

Which all of these should be addressed by separate task.